code/modules/spells/spell.dm

```diff
@@ -1,609 +1,454 @@
+ */
+/datum/action/cooldown/spell/proc/delevel_spell()
+	// Spell cannot be levelled
+	if(spell_max_level <= 1)
 		return FALSE
 
-	if(!user.can_cast_magic(antimagic_flags))
+	if(spell_level <= 1)
 		return FALSE
 
-	if(!ishuman(user))
-		if(clothes_req || human_req)
-			return FALSE
-		if(nonabstract_req && (isbrain(user) || ispAI(user)))
-			return FALSE
+	spell_level--
+	cooldown_time = min(cooldown_time + cooldown_reduction_per_rank, initial(cooldown_time))
+	update_spell_name()
 	return TRUE
 
-/obj/effect/proc_holder/spell/self //Targets only the caster. Good for buffs and heals, but probably not wise for fireballs (although they usually fireball themselves anyway, honke)
-	range = -1 //Duh
-
-/obj/effect/proc_holder/spell/self/choose_targets(mob/user = usr)
-	if(!user)
-		revert_cast()
-		return
-	perform(null,user=user)
-
-/obj/effect/proc_holder/spell/self/basic_heal //This spell exists mainly for debugging purposes, and also to show how casting works
-	name = "Lesser Heal"
-	desc = "Heals a small amount of brute and burn damage."
-	human_req = TRUE
-	clothes_req = FALSE
-	charge_max = 100
-	cooldown_min = 50
-	invocation = "Victus sano!"
-	invocation_type = INVOCATION_WHISPER
-	school = SCHOOL_RESTORATION
-	sound = 'sound/magic/staff_healing.ogg'
-
-/obj/effect/proc_holder/spell/self/basic_heal/cast(list/targets, mob/living/carbon/human/user) //Note the lack of "list/targets" here. Instead, use a "user" var depending on mob requirements.
-	//Also, notice the lack of a "for()" statement that looks through the targets. This is, again, because the spell can only have a single target.
-	user.visible_message(span_warning("A wreath of gentle light passes over [user]!"), span_notice("You wreath yourself in healing light!"))
-	user.adjustBruteLoss(-10)
-	user.adjustFireLoss(-10)
-
-/obj/effect/proc_holder/spell/vv_get_dropdown()
-	. = ..()
-	VV_DROPDOWN_OPTION("", "---------")
-	if(clothes_req)
-		VV_DROPDOWN_OPTION(VV_HK_SPELL_SET_ROBELESS, "Set Robeless")
-	else
-		VV_DROPDOWN_OPTION(VV_HK_SPELL_UNSET_ROBELESS, "Unset Robeless")
-
-	if(human_req)
-		VV_DROPDOWN_OPTION(VV_HK_SPELL_UNSET_HUMANONLY, "Unset Require Humanoid Mob")
-	else
-		VV_DROPDOWN_OPTION(VV_HK_SPELL_SET_HUMANONLY, "Set Require Humanoid Mob")
+/**
+ * Updates the spell's name based on its level.
+ */
+/datum/action/cooldown/spell/proc/update_spell_name()
+	var/spell_title = ""
+	switch(spell_level)
+		if(2)
+			spell_title = "Efficient "
+		if(3)
+			spell_title = "Quickened "
+		if(4)
+			spell_title = "Free "
+		if(5)
+			spell_title = "Instant "
+		if(6)
+			spell_title = "Ludicrous "
+
+	name = "[spell_title][initial(name)]"
+	UpdateButtons()
+
+/// Debug proc for seeing at a glance what all spells have as set requirements
+/// Probably remove before merge, idk it might be useful in the future as a debug verb.
+/proc/debug_spell_requirements()
+	for(var/datum/action/cooldown/spell/spell as anything in typesof(/datum/action/cooldown/spell))
```

Comment:
if you keep it, make it a debug verb (if not already)
also sum the text rather then just spamming messages